### PR TITLE
Improve viewport navigation 

### PR DESCRIPTION
**Pull request by Kiisu-Master**
Added new keyboard shortcuts and input setting.

If `use_ctrl_for_zoom` is off, scrolling will zoom in and out, holding ctrl while scrolling will pan the view up and down . Additionally holding shift changes it to pan left and right.

If `use_ctrl_for_zoom` is on, scrolling will pan up and down, holding ctrl while scrolling will zoom in and out. Only holding shift will pan left and right.

Also enabled smooth movement when using touchpad.

### CODE REVIEW

**Comment by MewPurPur** [comment]
Can you explain what this PR changes from user perspective? Like what the new functionalities are.

**Comment by Kiisu-Master** [comment]
Basically Godot 2d viewport navigation..
So keyboard shortcuts to change what scrolling and 2 finger panning on touchpad does.
Might break pinch zooming on touchpads because ctrl + scroll is now panning up and down.

**Comment by MewPurPur** [comment]
If you had to write exhaustive documentation on how to navigate in the 2D viewport after this PR, what would you write?

**Comment by Kiisu-Master** [comment]
These could be changed, but right now its like this:

Touchpad uses the factor to have precise movement without steps.
Pressing Ctrl enables panning with scrolling.
Pressing Shift and Ctrl swaps the panning directions.
- Zoom in/out:
  - Scroll up / down
  - Touchpad 2 finger up / down 
- Pan up / down:
  - Ctrl + Scroll up / down
  - Ctrl + Touchpad 2 finger up / down
- Pan left / right:
  - Ctrl + Shift + Scroll up / down
  - Touchpad 2 finger left / right

As i said before, pinch zooming on touchpad does not work for me anymore because it sends a ctrl + scroll event without factor, which just makes it pan up and down now. This used to work before because it zoomed on scroll events, no matter if ctrl is pressed or not.

File: src/ui_parts/viewport.gd [hunk 1]
```diff
@@ -57,21 +57,39 @@ func _unhandled_input(event: InputEvent) -> void:
...
-				zoom_menu.zoom_in()
-			MOUSE_BUTTON_WHEEL_DOWN:
-				zoom_menu.zoom_out()
-	
-	if event is InputEventMouseButton:
-		if event.ctrl_pressed:
-			pass
-
-
+		var fac = event.factor
```
Comment (MewPurPur) [inline src/ui_parts/viewport.gd:60]: Static typing here

File: src/ui_parts/zoom_menu.gd [hunk 2]
```diff
@@ -17,11 +17,20 @@ var zoom_level: float:
 		update_buttons_appearance()
 
 
-func zoom_out() -> void:
-	zoom_level /= sqrt(2)
-
-func zoom_in() -> void:
-	zoom_level *= sqrt(2)
+func zoom_out(factor: float = 1.0) -> void:
```
Comment (MewPurPur) [inline src/ui_parts/zoom_menu.gd:20]: The codebase generally uses `:=` when there's a default value.

File: src/ui_parts/viewport.gd [hunk 3]
```diff
@@ -57,39 +57,35 @@ func _unhandled_input(event: InputEvent) -> void:
...
-					zoom_menu.zoom_in(fac)
-				MOUSE_BUTTON_WHEEL_DOWN when GlobalSettings.invert_zoom:
-					zoom_menu.zoom_in(fac)
-				MOUSE_BUTTON_WHEEL_DOWN:
-					zoom_menu.zoom_out(fac)
+				MOUSE_BUTTON_WHEEL_UP: move_vec = Vector2.UP
+				MOUSE_BUTTON_WHEEL_DOWN: move_vec = Vector2.DOWN
+				MOUSE_BUTTON_WHEEL_LEFT: move_vec = Vector2.LEFT
+				MOUSE_BUTTON_WHEEL_RIGHT: move_vec = Vector2.RIGHT
+		var fac = event.factor
```
Comment (MewPurPur) [inline src/ui_parts/viewport.gd]: Add static typing here. Also, while at it, let's have 2 spaces between the code and the comment, consistent with the rest of the documentation.

Comment (Kiisu-Master) [reply src/ui_parts/viewport.gd]: Ill fix the spaces before the comment.
Idk how it happened but i think you just reviewed an older version of this PR.
I already added static typing to that.

**Comment by MewPurPur** [comment]
Needs a rebase, otherwise good.

**Comment by MewPurPur** [comment]
Thanks!

**Pull request merged**